packages/fiori/src/UploadCollectionItem.js

```diff
@@ -0,0 +1,184 @@
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { fetchI18nBundle, getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import Button from "@ui5/webcomponents/dist/Button.js";
+import ListItem from "@ui5/webcomponents/dist/ListItem.js";
+import {
+	UPLOADCOLLECTIONITEM_CANCELBUTTON_TEXT,
+	UPLOADCOLLECTIONITEM_RENAMEBUTTON_TEXT,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Template
+import UploadCollectionItemTemplate from "./generated/templates/UploadCollectionItemTemplate.lit.js";
+
+// Styles
+import UploadCollectionItemCss from "./generated/themes/UploadCollectionItem.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-upload-collection-item",
+	properties: /** @lends sap.ui.webcomponents.fiori.UploadCollectionItem.prototype */ {
+		file: {
+			type: Object,
+			defaultValue: null,
+		},
+		fileName: {
+			type: String,
+		},
+		fileNameClickable: {
+			type: Boolean,
+		},
+		noDelete: {
+			type: Boolean,
+		},
+		_editing: {
+			type: Boolean,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.fiori.UploadCollectionItem.prototype */ {
+		"default": {
+			type: Node,
+		},
+		thumbnail: {
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.fiori.UploadCollectionItem.prototype */ {
+		fileNameClick: { },
+		_rename: { },
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * For the <code>ui5-upload-collection-item</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents-fiori/dist/UploadCollectionItem.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.UploadCollectionItem
+ * @extends UI5Element
+ * @tagname ui5-upload-collection-item
+ * @public
+ */
+class UploadCollectionItem extends ListItem {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return [ListItem.styles, UploadCollectionItemCss];
+	}
+
+	static get template() {
+		return UploadCollectionItemTemplate;
+	}
+
+	static async onDefine() {
+		await Promise.all([
+			Button.define(),
+			fetchI18nBundle("@ui5/webcomponents"),
+		]);
+	}
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	onBeforeRendering() {
+		if (!this.focused) {
+			this._editing = false;
+		}
+	}
+
+	onAfterRendering() {
+		if (this.focused && this._editing) {
+			const inp = this.shadowRoot.getElementById("ui5-uci-edit-input");
```

Comment:
Can you use our declarative focus dom ref and call focus()?

Reply:
Done.